File: sdkwork-ai-sdk-go/api/images.go
package api

import (
    sdktypes "github.com/sdkwork/ai-sdk/types"
    sdkhttp "github.com/sdkwork/ai-sdk/http"
)

type ImagesApi struct {
    client *sdkhttp.Client
}

func NewImagesApi(client *sdkhttp.Client) *ImagesApi {
    return &ImagesApi{client: client}
}

// Create image variation
func (a *ImagesApi) CreateImageVariation(body *sdktypes.CreateImageVariationRequest, query map[string]interface{}) (sdktypes.ImageGenerationResponse, error) {
    raw, err := a.client.Post(AiApiPath("/images/variations"), body, query, nil, "multipart/form-data")
    if err != nil {
        var zero sdktypes.ImageGenerationResponse
        return zero, err
    }
    return decodeResult[sdktypes.ImageGenerationResponse](raw)
}

// Create image
func (a *ImagesApi) CreateImage(body sdktypes.ImageGenerationRequest) (sdktypes.ImageGenerationResponse, error) {
    raw, err := a.client.Post(AiApiPath("/images/generations"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.ImageGenerationResponse
        return zero, err
    }
    return decodeResult[sdktypes.ImageGenerationResponse](raw)
}

// Create image edit
func (a *ImagesApi) CreateImageEdit(body *sdktypes.CreateImageEditPostRequest, query map[string]interface{}) (sdktypes.ImageGenerationResponse, error) {
    raw, err := a.client.Post(AiApiPath("/images/edits"), body, query, nil, "multipart/form-data")
    if err != nil {
        var zero sdktypes.ImageGenerationResponse
        return zero, err
    }
    return decodeResult[sdktypes.ImageGenerationResponse](raw)
}

File: sdkwork-ai-sdk-typescript/src/api/images.ts
import { aiApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { ImageGenerationRequest, ImageGenerationResponse } from '../types';


export class ImagesApi2 {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** Create image variation */
  async createImageVariation(body?: FormData, params?: QueryParams): Promise<ImageGenerationResponse> {
    return this.client.post<ImageGenerationResponse>(aiApiPath(`/v1/images/variations`), body, params);
  }

/** Create image */
  async createImage(body: ImageGenerationRequest): Promise<ImageGenerationResponse> {
    return this.client.post<ImageGenerationResponse>(aiApiPath(`/v1/images/generations`), body);
  }

/** Create image edit */
  async createImageEdit(body?: FormData, params?: QueryParams): Promise<ImageGenerationResponse> {
    return this.client.post<ImageGenerationResponse>(aiApiPath(`/v1/images/edits`), body, params);
  }
}

export function createImagesApi2(client: HttpClient): ImagesApi2 {
  return new ImagesApi2(client);
}

export { ImagesApi2 as ImagesApi, createImagesApi2 as createImagesApi };
